soc: silabs: Add RAIL Multiprotocol support

loopj opened the pull request:
This PR adds support for the Silicon Labs RAIL Multiprotocol library, which provides coexistence and arbitration between multiple wireless protocols (for example, Bluetooth LE and a proprietary 2.4 GHz stack) on Gecko SoCs.

Depends on https://github.com/zephyrproject-rtos/hal_silabs/pull/126

loopj committed:
soc: silabs: Add Kconfig options to enable RAIL Multiprotocol support

Adds Kconfig options to enable RAIL Multiprotocol support

Signed-off-by: James Smith <james@loopj.com>

loopj committed:
modules: silabs: Link against RAIL Multiprotocol blobs when enabled

Enables RAIL Multiprotocol support by linking against the multiprotocol
RAIL library blob when `CONFIG_SILABS_GECKO_RAIL_MULTIPROTOCOL` or
`CONFIG_SILABS_SISDK_RAIL_MULTIPROTOCOL` are enabled.

Signed-off-by: James Smith <james@loopj.com>

asmellby reviewed (commented):
Nice, thanks a lot for this!

loopj committed:
modules: silabs: Fix broken binary blobs path with Gecko SDK

Bugfix to correct broken binary blobs path when using the Gecko SDK.

Signed-off-by: James Smith <james@loopj.com>

loopj committed:
modules: hal_silabs: Set multiprotocol define based on Kconfig

Set `SL_RAIL_LIB_MULTIPROTOCOL_SUPPORT` to `1` when Rail
Multiprotocol Kconfig is enabled.

This is required to run the Bluetooth multiprotocol initialization
code during `sl_btctrl_init`.

Signed-off-by: James Smith <james@loopj.com>

asmellby commented on modules/hal_silabs/gecko/Kconfig:
```diff
@@ -0,0 +1,16 @@
+config SILABS_GECKO_RAIL_MULTIPROTOCOL
+	bool "Use RAIL Multiprotocol library"
+	depends on SOC_GECKO_USE_RAIL
+	default n
```
Boolean kconfig options default to `n`, and shouldn't be explicitly set according to https://docs.zephyrproject.org/latest/build/kconfig/tips.html#redundant-defaults

asmellby commented on modules/hal_silabs/simplicity_sdk/config/ll/sl_btctrl_config.h:
```diff
@@ -123,6 +123,10 @@
 #endif
 
+#ifdef CONFIG_SILABS_SISDK_RAIL_MULTIPROTOCOL
+#define SL_RAIL_LIB_MULTIPROTOCOL_SUPPORT 1
```
This symbol seems to be set by the `rail_lib_(single|multi)protocol` components in SiSDK, and it is always explicitly set to `0` or `1` depending on which variant of the library is selected. Should this be set from CMakeLists.txt whenever RAIL is included, instead of in a Bluetooth-specific header? Other protocol stacks may depend on it too.

loopj commented:
@asmellby @jerome-pouiller can you please discuss and give me some concrete guidance on the following

1. If setting the `#define SL_RAIL_LIB_MULTIPROTOCOL_SUPPORT` explicitly to `0` or `1` (vs implicitly) need to be changed - I set these explicitly to mirror the `.slcc` behavior in `simplicity_sdk`
2. What I should name the `Kconfig` variable - I've already changed this once, so I guess I'll hold off until I have clarity

In the meantime I've rebased from main to avoid the west.yml conflict so we should be good to go there.

Thanks!

jerome-pouiller commented:
> * If setting the #define SL_RAIL_LIB_MULTIPROTOCOL_SUPPORT explicitly to 0 or 1 (vs implicitly) need to be changed.

Let's keep the current code until a real policy emerge.

> * What I should name the Kconfig variable - I've already changed this once, so I guess I'll hold off until I have clarity

The right thing to do is to rename the `RAIL_PA_` options. I will take care of this change today (I want to avoid half done clean-up).
